sklearn/metrics/cluster/tests/test_unsupervised.py

```diff
@@ -396,6 +398,172 @@ def test_davies_bouldin_score():
     pytest.approx(davies_bouldin_score(X, labels), (5.0 / 4) / 3)
 
 
+@pytest.fixture
+def density_sample():
+    points, labels = datasets.make_moons()
+    bounds = np.array([np.max(points, axis=0), np.min(points, axis=0)]).transpose()
+    noise = [
+        [np.random.uniform(*bounds[0]), np.random.uniform(*bounds[1])]
+        for _ in range(10)
+    ]
+    points = np.append(points, noise, axis=0)
+    labels = np.append(labels, [-1 for _ in range(10)])
+    return points, labels
+
+
+@pytest.fixture
+def density_samples_good_bad():
+    single_gauss_blob, _ = datasets.make_blobs(centers=1, random_state=42)
+    # normally distributed data, randomly split into two clusters
+    # (shuffled samples are labeled based on index mod 2)
+    bad = single_gauss_blob, [i % 2 for i in range(single_gauss_blob.shape[0])]
+    return datasets.make_moons(random_state=42), bad
+
+
+def test_dbcv_score_basic_validation_errs():
+    assert_raises_on_only_one_label(dbcv_score)
+    assert_raises_on_all_points_same_cluster(dbcv_score)
+
+
+def test_dbcv_score_precomputed_missing_d_valerr(density_sample):
+    msg = "If metric is precomputed a d value must be provided!"
+    with pytest.raises(ValueError, match=msg):
+        dbcv_score(
+            pairwise_distances(density_sample[0]),
+            density_sample[1],
+            metric="precomputed",
+        )
+
+
+def test_dbcv_score_irrelevant_d_warning(density_sample):
+    expected_msg = (
+        'The "d" value you provided is being ignored. '
+        "It's only required for precomputed distances"
+    )
+    with pytest.warns(UserWarning, match=expected_msg):
+        dbcv_score(*density_sample, d=2)
+
+
```

Comment:
Can we parametrize the following two tests by some other metrics as well to confirm they work as expected?

Reply:
I meant currently I think we are only testing using the euclidean distance metric as that is the default. Maybe we can include a couple of other metrics like manhattan, Minkowski e.t.c as well which can be easily included using a pytest.parametrize decorator.

Reply:
Oh ok, thank you for the clarification. Tonight, I'll look into it and throw something together.

Reply:
ok: [b266ffb](https://github.com/luis261/scikit-learn/commit/b266ffb91cc639a6c13fbb12cacd74fd8a79cc42)

tests pass, just gotta fix the new linting issue and we should be good to go